hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/debug/ReadReplicas.java

```diff
@@ -91,64 +93,72 @@ public Class<?> getParentType() {
+      String bucketName = address.getBucketName();
+      String keyName = address.getKeyName();
 
-    String directoryName = createDirectory(volumeName, bucketName, keyName);
+      File dir = createDirectory(volumeName, bucketName, keyName);
 
-    OzoneKeyDetails keyInfoDetails
-        = clientProtocol.getKeyDetails(volumeName, bucketName, keyName);
+      OzoneKeyDetails keyInfoDetails
+          = checksumClient.getKeyDetails(volumeName, bucketName, keyName);
 
-    Map<OmKeyLocationInfo, Map<DatanodeDetails, OzoneInputStream>> replicas
-        = clientProtocol.getKeysEveryReplicas(volumeName, bucketName, keyName);
+      Map<OmKeyLocationInfo, Map<DatanodeDetails, OzoneInputStream>> replicas =
+          checksumClient.getKeysEveryReplicas(volumeName, bucketName, keyName);
 
-    Map<OmKeyLocationInfo, Map<DatanodeDetails, OzoneInputStream>>
-        replicasWithoutChecksum = clientProtocolWithoutChecksum
-        .getKeysEveryReplicas(volumeName, bucketName, keyName);
+      Map<OmKeyLocationInfo, Map<DatanodeDetails, OzoneInputStream>>
+          replicasWithoutChecksum = noChecksumClient
+          .getKeysEveryReplicas(volumeName, bucketName, keyName);
 
-    JsonObject result = new JsonObject();
-    result.addProperty(JSON_PROPERTY_FILE_NAME,
-        volumeName + "/" + bucketName + "/" + keyName);
-    result.addProperty(JSON_PROPERTY_FILE_SIZE, keyInfoDetails.getDataSize());
+      JsonObject result = new JsonObject();
+      result.addProperty(JSON_PROPERTY_FILE_NAME,
+          volumeName + "/" + bucketName + "/" + keyName);
+      result.addProperty(JSON_PROPERTY_FILE_SIZE, keyInfoDetails.getDataSize());
 
-    JsonArray blocks = new JsonArray();
-    downloadReplicasAndCreateManifest(keyName, replicas,
-        replicasWithoutChecksum, directoryName, blocks);
-    result.add(JSON_PROPERTY_FILE_BLOCKS, blocks);
+      JsonArray blocks = new JsonArray();
+      downloadReplicasAndCreateManifest(keyName, replicas,
+          replicasWithoutChecksum, dir, blocks);
+      result.add(JSON_PROPERTY_FILE_BLOCKS, blocks);
 
-    Gson gson = new GsonBuilder().setPrettyPrinting().create();
-    String prettyJson = gson.toJson(result);
+      Gson gson = new GsonBuilder().setPrettyPrinting().create();
+      String prettyJson = gson.toJson(result);
 
-    String manifestFileName = keyName + "_manifest";
-    System.out.println("Writing manifest file : " + manifestFileName);
-    File manifestFile
-        = new File(outputDir + "/" + directoryName + "/" + manifestFileName);
-    Files.write(manifestFile.toPath(),
-        prettyJson.getBytes(StandardCharsets.UTF_8));
+      String manifestFileName = keyName + "_manifest";
+      System.out.println("Writing manifest file : " + manifestFileName);
+      File manifestFile
+          = new File(dir, manifestFileName);
+      Files.write(manifestFile.toPath(),
+          prettyJson.getBytes(StandardCharsets.UTF_8));
+    } finally {
+      newClient.close();
```

Comment:
Can this finally block be moved to try-with-resource. Since if future change updates this inside try block, it might lead to same problem.

Reply:
Sorry, I don't get it what kind of future change inside the try block would lead to the same problem.

Reply:
Oh I meant if the newClient variable is reassigned another value due to some other change, it might still lead to the initial value newClient not being closed. In this case try(newClient) would be better & would ensure we are closing the stream we opened.